__tests__/pet.test.js

```diff
@@ -0,0 +1,218 @@
+    });
+    it('throws an error if the pet is not alive', () => {
+      pet.age = 30;
+      expect(() => pet.feed()).toThrow('Your pet is no longer alive :(');
+    });
+  });
+
+  describe('checkUp', () => {
+    let pet;
+    beforeEach(() => {
+      pet = new Pet('Jessie');
+    });
+    it('returns I need to walk if pets fitness is 3 or less', () => {
+      pet.fitness = 3;
+      pet.checkUp();
+      expect(pet.checkUp()).toEqual('I need a walk')
+    });
+    it('returns I need to walk if pets fitness is 1', () => {
+      pet.fitness = 1;
+      pet.checkUp();
+      expect(pet.checkUp()).toEqual('I need a walk')
+    });
+    it('returns I am hungry if pets hunger is 5 or more', () => {
+      pet.hunger = 5;
+      pet.checkUp();
+      expect(pet.checkUp()).toEqual('I am hungry');
+    });
+    it('returns I am hungry if pets hunger is 9', () => {
+      pet.hunger = 9;
+      pet.checkUp();
+      expect(pet.checkUp()).toEqual('I am hungry');
+    });
+    it('returns I need a walk AND I am hungry if fitness is 3 or less and its hunger is 5 or more', () => {
+      pet.fitness = 3;
+      pet.hunger = 5;
+      expect(pet.checkUp()).toEqual('I need a walk AND I am hungry');
+    });
+    it('returns I need a walk AND I am hungry if fitness is 1 and its hunger is 8', () => {
+      pet.fitness = 1;
+      pet.hunger = 8;
+      expect(pet.checkUp()).toEqual('I need a walk AND I am hungry');
+    });
+    it('returns I feel great if pets fitness is 4 or above and its hunger is 4 or less', () => {
+      pet.fitness = 4;
+      pet.hunger = 4;
+      expect(pet.checkUp()).toEqual('I feel great');
+    });
+    it('returns I feel great if pets fitness is 7 and its hunger is 2', () => {
+      pet.fitness = 7;
+      pet.hunger = 2;
+      expect(pet.checkUp()).toEqual('I feel great');
+    });
+    it('throws an error if the pet is not alive', () => {
+      pet.age = 30;
+      expect(() => pet.checkUp()).toThrow('Your pet is no longer alive :(');
+    });
+  });
+
+  describe('isAlive', () => {
+    let pet;
+    beforeEach(() => {
+      pet = new Pet('Jessie');
+    });
+    it('returns false if fitness is 0 or less', () => {
+      pet.fitness = 0;
+      expect(pet.isAlive).toBe(false);
+    });
+    it('returns false if hunger is 10 or more', () => {
+      pet.hunger = 10;
+      expect(pet.isAlive).toBe(false);
+    });
+    it('returns false if pet age is 30 or more', () => {
+      pet.age = 30;
+      expect(pet.isAlive).toBe(false);
+    });
+    it('returns true if pet fitness is 1 or above, hunger is 9 or less, age is under 30', () => {
+      pet.fitness = 1;
+      pet.hunger = 9;
+      pet.age = 29;
+      expect(pet.isAlive).toBe(true);
+    });
+  });
+
+  describe('adoptChild', () => {
+    let parent;
+    let child;
+    beforeEach(() => {
+      parent = new Pet('Jessie');
+      child = new Pet('Nima');
+    });
+    it('adds child to parent property', () => {
+      parent.adoptChild(child);
```

Comment:
Instead of adding this line to every `it` block, since it is being used in all of them could it be moved into the `beforeEach` instead?